tools/validators-monitor/notification/notifier.go

```diff
@@ -0,0 +1,204 @@
+package notification
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"math/big"
+	"net/http"
+	"time"
+
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/primev/mev-commit/tools/validators-monitor/api"
+)
+
+// Message represents a notification message structure
+// generalized to support different platforms.
+type Message struct {
+	Text        string       `json:"text,omitempty"`
+	Attachments []Attachment `json:"attachments,omitempty"`
+}
+
+// Attachment represents a message attachment
+type Attachment struct {
+	Color      string   `json:"color,omitempty"`
+	Title      string   `json:"title,omitempty"`
+	Text       string   `json:"text,omitempty"`
+	Fields     []Field  `json:"fields,omitempty"`
+	Footer     string   `json:"footer,omitempty"`
+	TS         int64    `json:"ts,omitempty"`
+	MarkdownIn []string `json:"mrkdwn_in,omitempty"`
+}
+
+// Field represents a field in a message attachment
+type Field struct {
+	Title string `json:"title"`
+	Value string `json:"value"`
+	Short bool   `json:"short"`
+}
+
+// Notifier sends notifications to multiple webhook endpoints
+type Notifier struct {
+	webhookURLs []string
+	client      *http.Client
+	logger      *slog.Logger
+	enabled     bool
+}
+
+// NewNotifier creates a new notifier instance
+func NewNotifier(webhookURLs []string, logger *slog.Logger) *Notifier {
+	enabled := len(webhookURLs) > 0
+
+	if !enabled {
+		logger.Warn("Notifications disabled - no webhook URLs provided")
+	} else {
+		logger.Info("Notifications enabled")
+	}
+
+	return &Notifier{
+		webhookURLs: webhookURLs,
+		client: &http.Client{
+			Timeout: 10 * time.Second,
+		},
+		logger:  logger,
+		enabled: enabled,
+	}
+}
+
+// SendMessage sends a message to all configured webhook endpoints
+func (n *Notifier) SendMessage(ctx context.Context, message Message) error {
+	if !n.enabled {
+		n.logger.Debug("Notification skipped (disabled)")
+		return nil
+	}
+
+	messageJSON, err := json.Marshal(message)
+	if err != nil {
+		return fmt.Errorf("failed to marshal message: %w", err)
+	}
+
+	var errorOccurred bool
+	for _, webhookURL := range n.webhookURLs {
+		req, err := http.NewRequestWithContext(ctx, "POST", webhookURL, bytes.NewBuffer(messageJSON))
+		if err != nil {
+			n.logger.Error("Failed to create request", "webhook", webhookURL, "error", err)
+			errorOccurred = true
```

Comment:
Instead of setting this you can use errors.Join to collect all errors and then print them out for better information on failures.